src/main/java/com/pmi/SchemaCompiler/ConfigGeneratorMojo.java

```diff
@@ -33,6 +35,39 @@
   private static String pattern_smaato = "\\d{9}";
   private static String pattern_aps = "/21839579524/aps/amazon-(ios|android)-.+";
 
+  private static Comparator<Map<String, Object>> comparator =
+      new Comparator<Map<String, Object>>() {
+
+        @Override
+        public int compare(Map<String, Object> o1, Map<String, Object> o2) {
+          if (!o1.containsKey("price") && !o2.containsKey("price")) {
+            return 0;
+          } else if (!o1.containsKey("price")) {
+            return -1;
+          } else if (!o2.containsKey("price")) {
+            return 1;
+          } else {
+            double price1 = new Double(o1.get("price").toString());
+            double price2 = new Double(o2.get("price").toString());
+            if ("ad_fb_native".equals(o1.get("ctype")) && price1 == 0) {
+              return -1;
+            }
+
+            if ("ad_fb_native".equals(o2.get("ctype")) && price2 == 0) {
+              return 1;
+            }
+
+            if (price1 > price2) {
```

Comment:
Double already has built-in compareTo or compare

Reply:
Fixed.